### PR TITLE
Create a new module for Azure SQL Failover Group and add the first parent service "azure-sql-12-0-dr-dbms-pair-registered"

### PR DESCRIPTION
**Pull request by zhongyi-zhang**
Kick off the failover group part in https://github.com/Azure/open-service-broker-azure/issues/566!

### CODE REVIEW

File: pkg/services/mssqldr/catalog.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+				Metadata: service.ServiceMetadata{
+					DisplayName:      "Azure SQL 12.0-- disaster recovery DBMS Pair registered",
+					ImageURL:         "https://azure.microsoft.com/svghandler/sql-database/?width=200",
+					LongDescription:  "Azure SQL 12.0-- disaster recovery DBMS pair registered, as the primary server and the secondary server of failover groups",
+					DocumentationURL: "https://docs.microsoft.com/en-us/azure/sql-database/",
+					SupportURL:       "https://azure.microsoft.com/en-us/support/",
+				},
+				Bindable: false,
+				Tags: []string{
+					"AzurePublicCloud",
```
Comment (norshtein) [inline pkg/services/mssqldr/catalog.go]: I think tags `Azure*Cloud` should be removed.

**Comment by norshtein** [comment]
Better to add a doc in the PR. If you want to add a summarized doc after several PRs are merged, please add explanation about what's the functionality of `dbmsPairRegisteredManager`.

File: pkg/services/mssqldr/catalog.go (hunk 1 above)
Comment (zhongyi-zhang) [reply pkg/services/mssqldr/catalog.go]: Yeah, thanks for the good catch.

**Comment by zhongyi-zhang** [comment]
Sure. Will add a comment for `dbmsPairRegisteredManager`.

**Comment by norshtein** [comment]
It's better to add the behavior of provision/bind/unbind/deprovision/update

**Comment by zhongyi-zhang** [comment]
OK, I added a draft doc instead for the overview. Will continue polishing it according to the reviews during addressing the services in this module.

File: pkg/services/mssqldr/common.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	"net/url"
+
+	_ "github.com/denisenkom/go-mssqldb" // MS SQL Driver
+)
+
+func getDBConnection(
+	administratorLogin string,
+	administratorLoginPassword string,
+	fullyQualifiedDomainName string,
+	databaseName string, // nolint: unparam
```
Comment (norshtein) [inline pkg/services/mssqldr/common.go]: Why use `nolint: unparam` here? Parameter `databaseName` is used in the function and shouldn't cause a lint issue.

Comment (zhongyi-zhang) [reply pkg/services/mssqldr/common.go]: The linter reports a warning when a parameter always receives the same value in a function. Now the child services are not included in the module, the value of `databaseName` is always `master`.

File: pkg/services/mssqldr/dbms_pair_registered_update.go [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+) error {
+	return nil
+}
+
+func (d *dbmsPairRegisteredManager) GetUpdater(
+	service.Plan,
+) (service.Updater, error) {
+	return service.NewUpdater(
+		service.NewUpdatingStep("updateAdministrators", d.updateAdministrators),
+		service.NewUpdatingStep("testPriConnection", d.testPriConnection),
```
Comment (norshtein) [inline pkg/services/mssqldr/dbms_pair_registered_update.go]: Here exists the same potential bug as I raised in #570 . The user may provide wrong password by mistake, which will cause step testConnection fails, but the wrong parameter will be written to store in step updateAdministrator and previous correct username and password will be overwritten.

File: pkg/services/mssqldr/mssqldr.go [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+// New returns a new instance of a type that fulfills the service.Module
+// interface and is capable of provisioning MS SQL servers and databases
+// using "Azure SQL Database"
+func New(
+	azureEnvironment azure.Environment,
+	armDeployer arm.Deployer,
+	serversClient sqlSDK.ServersClient,
+	databasesClient sqlSDK.DatabasesClient, // nolint: unparam
```
Comment (norshtein) [inline pkg/services/mssqldr/mssqldr.go]: I guess parameter `databasesClient` and `failoverGroupsClient` be used in later submitted PRs? Better to add a comment here.

File: pkg/services/mssqldr/plan_details.go [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+			"secondaryServer",
+			"secondaryAdministratorLogin",
+			"secondaryAdministratorLoginPassword",
+		},
+		SecureProperties: []string{
+			"primaryAdministratorLoginPassword",
+			"secondaryAdministratorLoginPassword",
+		},
+		PropertySchemas: map[string]service.PropertySchema{
+			"primaryResourceGroup": &service.StringPropertySchema{
```
Comment (norshtein) [inline pkg/services/mssqldr/plan_details.go]: `Title` is missed, the same in below schemas.

File: pkg/services/mssqldr/common.go (hunk 2 above)
Comment (norshtein) [reply pkg/services/mssqldr/common.go]: Thanks for the explanation!

File: pkg/services/mssqldr/dbms_pair_registered_update.go (hunk 3 above)
Comment (zhongyi-zhang) [reply pkg/services/mssqldr/dbms_pair_registered_update.go]: Yes, I almost forgot it. Thanks for raising it again.

**Comment by zhongyi-zhang** [comment]
Got error "Microsoft Internal subscriptions are restricted from provisioning in this region. Please choose a different region.".  Tried another two and rerun now.

**Comment by zhongyi-zhang** [comment]
@krancour @jeremyrickard could you help get another available location for the CI subscription to provision SQL Servers? Southcentralus is available. The cases in this module need one more. I can only guess and try...

**Comment by zhongyi-zhang** [comment]
It is only Microsoft Internal subscriptions constraint. External users won't run into it.

**Comment by norshtein** [comment]
Got it. Thanks for the explanation.

**Pull request merged**